src/js/Form.js

```diff
@@ -388,13 +388,20 @@ define( function( require, exports, module ) {
             },
             next: function() {
                 var next, currentIndex;
+                var that = this;
                 this.updateAllActive();
                 currentIndex = this.getCurrentIndex();
-                next = this.getNext( currentIndex );
 
-                if ( next ) {
-                    this.flipTo( next, currentIndex + 1 );
-                }
+                form.validateContent( this.$current )
+                    .then( function( valid ) {
+                        if ( !valid ) return;
+
+                        next = that.getNext( currentIndex );
+
+                        if ( next ) {
+                            that.flipTo( next, currentIndex + 1 );
+                        }
+                    } );
```

Comment:
This bit ^^ is the new functionality - everything else is just refactoring and fiddling.